src/CdrReader.ts

```diff
@@ -260,21 +267,33 @@ export class CdrReader {
     this.origin = this.offset;
   }
 
-  /** Reads the PID_SENTINEL value if encapsulation kind supports it (PL_CDR version 1)*/
-  sentinelHeader(): void {
+  /** Reads boolean flag for optional members in CDR2
+   * Will throw an error if called for CDR1.
+   */
+  isPresentFlag(): boolean {
+    if (!this.isCDR2) {
+      throw new Error("isPresentFlag is only supported for CDR2");
+    }
+    const isPresent = Boolean(this.uint8());
+    return isPresent;
+  }
+
+  /** Reads the PID_SENTINEL value if encapsulation kind supports it (PL_CDR version 1)
+   * @returns true if the sentinel header was read, false otherwise
+   */
+  sentinelHeader(): boolean {
```

Comment:
maybe name it like `maybeSentinelHeader` to be clearer that it may or may not work?

Reply:
I don't really want to rename it since it's a bit of a breaking change :/